Fixed extra characters in Chinese input

nanshiki opened the pull request:
Fixed #4227.
Fixed extra characters in input when IME is enabled by setting codepage in [dos] country instead of [dosv] dosv setting.

nanshiki commented:
Click on Actions above and select the build from the list "Merge pull request #4253 from nanshiki/fiximeinput" that matches your environment.

Arthur3032 commented:
Where can I download the version that fixes this problem? (DosBox-X.exe or install file)

nanshiki commented:
Click on [Actions](https://github.com/joncampbell123/dosbox-x/actions) to see the list and click on the title of the builds that match your environment.
There is no nanshiki/fiximeinput on the first page, but it can be any of them since they have already been merged.
If you are in a Windows environment, you can use 64-bit Visual Studio builds or 32-bit Visual Studio builds, and the installer is Visual Studio builds for WinXP.
There is a link to the file under Artifacts, so download it.

Arthur3032 commented:
Can't find it yet, can you give me the download link?
(Visual Studio Win 64bit)

Arthur3032 commented:
It has been used normally. Thank you very much!

Arthur3032 commented:
Can you give 32Bit a link?

Arthur3032 commented:
This virsion "Win64 "
Can inkput Chinese, but Can't keyin ABCDEF....

nanshiki commented:
I tried to correct it.
Only dosbox-x.exe for win64 SDL1 version. Please replace it and try it.
[dosbox-x-test.zip](https://github.com/joncampbell123/dosbox-x/files/11468735/dosbox-x-test.zip)

Arthur3032 commented:
ok, very good. Thank you.

1abcd commented:
I found that pressing shift key with number to input a chinese phrase by using 微軟倉頡 (MS ChangJie IME) still have extra character appeared.

<img width="511" alt="未命名" src="https://github.com/joncampbell123/dosbox-x/assets/1331388/8073e1fe-2d9d-4204-879d-44ebb8317a67">

The DOSBox-x [Development (preview/testing) builds] is downloaded from https://dosbox-x.com/

nanshiki commented:
Will this occur with #4267?
How about with Version 2022.12.26?

1abcd commented:
>Will this occur with #4267

yes

>How about with Version 2022.12.26?

the same

>Only dosbox-x.exe for win64 SDL1 version. Please replace it and try it.
[dosbox-x-test2.zip](https://github.com/joncampbell123/dosbox-x/files/11489592/dosbox-x-test2.zip)

I try this one. But I am not sure, it seems more hard to type digits after chinese typing.
I often use bopomofo IME, as many Taiwan people do, so I cant find other IME problems often.

nanshiki commented:
I installed MS ChangJie IME on a Japanese OS and used this test version, and confirmed that no extra symbols were entered with Shift + number at the next guess candidate.
In your environment, is the symbol entered with Shift + number in this test version?
![ime](https://github.com/joncampbell123/dosbox-x/assets/42603716/929d570d-33a1-48eb-91ca-1bf553a6b4e4)

1abcd commented:
> In your environment, is the symbol entered with Shift + number in this test version?

Thanks. It was fixed. No extra symbol appeared.

But in my envirement, sometimes digital characters (1234567890) was replaced by other symbol characters (!@#$%^&*()).
I don't known what caused the problem.
<img width="492" alt="未命名" src="https://github.com/joncampbell123/dosbox-x/assets/1331388/20882c0e-7bf7-4a7a-aee0-af7041b6d372">

nanshiki commented:
I found the root cause and fixed it again.
Please try this test4 instead of test3.
[dosbox-x-test4.zip](https://github.com/joncampbell123/dosbox-x/files/11509050/dosbox-x-test4.zip)

1abcd commented:
> I found the root cause and fixed it again. Please try this test4 instead of test3. [dosbox-x-test4.zip](https://github.com/joncampbell123/dosbox-x/files/11509050/dosbox-x-test4.zip)


<img src="https://user-images.githubusercontent.com/42603716/238975923-929d570d-33a1-48eb-91ca-1bf553a6b4e4.png">
On this screen, press any key of 1234567890 without shift key.
And change IME mode to 英 (press shift key in ChangJie), the IME stopped work.

nanshiki commented:
I have confirmed the problem and fixed it again.
This time I fixed the process on the SDL side and I think it is ok.
[dosbox-x-test5.zip](https://github.com/joncampbell123/dosbox-x/files/11515587/dosbox-x-test5.zip)

Arthur3032 commented:
test5.zip
The DoxBox-X debugger will appear.
how to make it not appear

1abcd commented:
> I have confirmed the problem and fixed it again. This time I fixed the process on the SDL side and I think it is ok. [dosbox-x-test5.zip](https://github.com/joncampbell123/dosbox-x/files/11515587/dosbox-x-test5.zip)

Test5 is ok.

nanshiki commented:
@Arthur3032 
Do you mean that the debugger screen appears when it is started?
This does not occur in our environment.
Is there something registered in [autoexec]?
Please try to continue with F5.
@1abcd 
Since test5 is OK, I will submit a pull request after I have verified it for a while longer.

Arthur3032 commented:
Instead of the debugger appearing when starting Dosbox-X,
Instead, when executing Doxbox-X l executes an external Win32 exe file.

Arthur3032 commented:
![擷取](https://github.com/joncampbell123/dosbox-x/assets/93595854/768c48d7-c6a1-4963-a16a-e0b5a830731b)

nanshiki commented:
@Arthur3032 
Fix #4299
